src/drivers/batt_smbus/batt_smbus.cpp

```diff
@@ -381,11 +590,53 @@ BATT_SMBUS::cycle()
 	uint64_t now = hrt_absolute_time();
 
 	// exit without rescheduling if we have failed to find a battery after 10 seconds
-	if (!_enabled && (now - _start_time > BATT_SMBUS_TIMEOUT_MS)) {
+	if (!_enabled && (now - _start_time > BATT_SMBUS_TIMEOUT_US)) {
 		warnx("did not find smart battery");
 		return;
 	}
 
+	bool perform_solo_battry_check = false; // Only check if it is a solo battery if changes have been made to the SBS data
+
+	// Try and get battery SBS info
+	if (_manufacturer_name == nullptr) {
+		char man_name[21];
+		uint8_t len = manufacturer_name((uint8_t *)man_name, sizeof(man_name));
+
+		if (len > 0) {
+			_manufacturer_name = new char[len];
+			strcpy(_manufacturer_name, man_name);
+			perform_solo_battry_check = true;
+		}
+	}
+
+	if (_device_name == nullptr) {
+		char dev_name[21];
+		uint8_t len = device_name((uint8_t *)dev_name, sizeof(dev_name));
+
+		if (len > 0) {
+			_device_name = new char[len];
+			strcpy(_device_name, dev_name);
+			perform_solo_battry_check = true;
+		}
+	}
+
+	if (_device_chemistry == nullptr) {
+		char dev_chem[21];
+		uint8_t len = device_chemistry((uint8_t *)dev_chem, sizeof(dev_chem));
+
+		if (len > 0) {
+			_device_chemistry = new char[len];
+			strcpy(_device_chemistry, dev_chem);
+			perform_solo_battry_check = true;
+		}
+	}
+
+	// If necessary, check if the battery is a 3DR Solo Battery
+	if (perform_solo_battry_check) {
+		warnx("Checking solo battery");
+		check_if_solo_battery();
+	}
```

Comment:
Here I try and read out the Manufacturer Name, Device Name and Device Chemistry. Should this be in cycle(), or should it rather be moved to the start function, since it really has to be only run once?